worldedit-core/src/main/java/com/sk89q/worldedit/world/registry/BundledBlockData.java

```diff
@@ -87,6 +87,64 @@ private void loadFromResource() throws IOException {
             entry.postDeserialization();
             idMap.put(entry.id, entry);
             legacyMap.put(entry.legacyId, entry);
+            if (entry.states == null) {
+                return;
+            }
+            SimpleState half = entry.states.get("half");
+            if (half != null && half.valueMap() != null) { // Fixes rotation for slabs and other half blocks
+                SimpleStateValue top = half.valueMap().get("top");
+                SimpleStateValue bot = half.valueMap().get("bottom");
+                if (top != null && top.getDirection() == null) {
+                    top.setDirection(new Vector(0, 1, 0));
+                }
+                if (bot != null && bot.getDirection() == null) {
+                    bot.setDirection(new Vector(0, -1, 0));
+                }
+                return;
+            }
+            SimpleState dir = entry.states.get("rotation");
+            if (dir != null && dir.valueMap() != null) {
+                Vector[] range = new Vector[]{new Vector(0, 0, -1),
```

Comment:
you can probably move this into a more permanent object instead of re-initializing it for each state. since vectors are immutable there's no danger of overwriting them.

on the other hand, i'm not sure how many blocks this will even apply to.